src/dynarec/arm64/dynarec_arm64_00.c

```diff
@@ -3177,6 +3176,14 @@ uintptr_t dynarec64_00(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nin
                     else
                         j64 = addr+i32;
                     jump_to_next(dyn, j64, 0, ninst, rex.is32bits);
+                    if(box64_dynarec_callret && addr >= (dyn->start + dyn->isize)) {
```

Comment:
I don't think this block is usefull. Why not simply jump to the `jump_to_next` that is just generated before?

Reply:
It's there for the `RET` to return.

Before this patch, the out-of-block jump "returned" to the jump table, this violates the RAS assumption and hurts the performance as I tested.

With this patch, the out-of-block jump "returned" to the end of `jump_to_next`, right after the `BLR`, so it's native PC + 4, same as what's in the LR register, which matches the RAS assumption.

Reply:
Actually, maybe the whole test on addr out of the block is useless, it should just return there to either continue or jump_to_next?!

Reply:
Oh are you saying that `&& addr >= (dyn->start + dyn->isize)` is unnecessary? well, yeah but it saves some memory if that matters...

Reply:
well, code compacity also helps to stay more in instruction cache... and if it's useless, it's useless..

Reply:
Sure, I'll remove it.